kolibri/core/assets/src/views/ContentRenderer/ContentRendererLoading.vue

```diff
@@ -0,0 +1,16 @@
+<template>
+
+  <KCircularLoader
+    :delay="false"
+  />
+
+</template>
+
+
+<script>
+
+  export default {
+    name: 'ContentRendererLoading',
+  };
+
+</script>
```

Comment:
couldn't `ContentRendererError` and `ContentRendererLoading` just be private sub-components of `KContentRenderer`?

Also, `ContentRendererLoading` seems maybe unnecessary since it doesn't add anything to `KCircularLoader`?

Reply:
I have them separately to be able to use them as part of the async definition pattern https://vuejs.org/v2/guide/components-dynamic-async.html#Handling-Loading-State - they are used outside of  `KContentRenderer` to render the loading state while renderer is loaded asynchronously and to show the error state if it fails to load.

I could probably define the loading component inline, as it is only used in the loading state handling, but I needed to set the props on the `KCircularLoader` - I could have used Vue.extend to extend the `KCircularLoader` component and set the default prop value to what I wanted here, but this seemed more explicit.

The final reason that the loading and error components are injected, rather than being part of the exported content renderer is to allow for an alternative (e.g. vuetify) loader to be used in place in Studio, or in the case of the error component for a different wrapper component (not UiAlert, which I believe we don't export) and different interface text.

Reply:
I'm still not clear on why they live in `core` rather than `kolibri-components` though? I would have expected something like

```
src/
  ...
  KContentRenderer/
    index.js
    ContentRendererError.vue
    ContentRendererLoading.vue
  ...
```

> The final reason ... is to allow for an alternative (e.g. vuetify) loader to be used in place in Studio, or in the case of the error component for a different wrapper component (not UiAlert, which I believe we don't export) and different interface text.

In theory at least our K loaders and the V loaders should look interchangeable, and we'll eventually expose a KUIAlert. In practice .. not quite there yet

Reply:
Right - because they were not both directly exportable, I just decided to go for the injection instead, can also add them as the 'default' but seems odd to add a default that does not work out of the box (in the case of the error component).